Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM amazonlinux:2017.03.1.20170812
+
+# lock yum to the same repository version
+RUN sed -i 's/releasever=.*/releasever=2017.03/g' /etc/yum.conf
+
+# base requirements
+RUN yum install yum-utils zip -y && \
+    yum-config-manager --enable epel && \
+    yum install wget git libpng-devel libcurl-devel gcc python27-devel libjpeg-devel -y
+
+# enable epel on Amazon Linux 2
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+
+# ImageMagick
+RUN yum install -y ImageMagick-devel
+
+# Other libraries
+RUN yum install -y pngcrush libjpeg* gifsicle
+
+# optipng
+RUN wget http://dl.fedoraproject.org/pub/epel/6/x86_64/Packages/o/optipng-0.7.6-6.el6.x86_64.rpm && \
+    yum localinstall optipng-0.7.6-6.el6.x86_64.rpm -y && rm optipng*rpm
+
+# pngquant
+RUN wget http://dl.fedoraproject.org/pub/epel/6/x86_64/Packages/l/libimagequant-2.5.2-5.el6.x86_64.rpm && \
+    yum localinstall libimagequant-2.5.2-5.el6.x86_64.rpm -y && rm libimagequant*rpm && \
+    wget http://dl.fedoraproject.org/pub/epel/6/x86_64/Packages/p/pngquant-2.5.2-5.el6.x86_64.rpm && \
+    yum localinstall pngquant-2.5.2-5.el6.x86_64.rpm -y && rm pngquant*rpm
```

Comment:
I was not able to find `optipng` or `pngquant` in `yum search`. In order to get around errors in including those when running the build script, I grabbed them directly from here. Now when I run it doesn't complain in the build script at the end.

```
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip pngquant
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip jpegtran
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip optipng
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip pngcrush
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip gifsicle
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip mozjpeg
zip -q -g /lambda/deployment/dist/env/../serverless-image-handler.zip imgmin
```

When I didn't have these dependencies, the output would say something to the effect of `nothing to do here` (when it couldn't find the dependencies).